src/main/java/com/google/sps/servlets/utils/Statistics.java

```diff
@@ -15,93 +15,137 @@
    *
-   * @param sentimentScores a list of score values from -1.0 to 1.0
+   * @param userCommentList a list of userComment analyzed from sentiment analysis with upadted
+   *     score and magnitude
+   * @param lowerEnd lower end boundary of the map
+   * @param upperEnd upper end boundary of the map
+   * @param interval interval for the range
+   * @return a categorized map based on userCommentList from lowerEnd to upperEnd with interval
    */
-  private void setAggregateScores(List<Double> sentimentScores) {
-    aggregateValues = new HashMap<>();
+  private List<SentimentBucket> categorizeToBucketList(
+      List<UserComment> userCommentList,
+      BigDecimal lowerEnd,
+      BigDecimal upperEnd,
+      BigDecimal interval,
+      int topNumComments) {
+    List<SentimentBucket> sentimentBucketList = new ArrayList<>();
     // Add score's interval to different ranges two sorting with and two pointers pop-up
-    sentimentScores.sort(Comparator.naturalOrder());
+    userCommentList.sort(ascendingScoreCompare);
     int updatingScoreIdx = 0;
-    for (BigDecimal tempPoint = LOWER_END;
-        tempPoint.compareTo(UPPER_END) < 0;
-        tempPoint = tempPoint.add(INTERVAL)) {
-      BigDecimal nextPoint = UPPER_END.min(tempPoint.add(INTERVAL));
+    BigDecimal tempPoint;
+    for (tempPoint = lowerEnd;
+        tempPoint.compareTo(upperEnd) < 0;
+        tempPoint = tempPoint.add(interval)) {
+      BigDecimal nextPoint = upperEnd.min(tempPoint.add(interval));
       Range currentRange = new Range(tempPoint, nextPoint);
-      aggregateValues.put(currentRange, 0);
+      int currentFrequency = 0;
+      PriorityQueue<UserComment> highMagnitudeComments = new PriorityQueue<>();
       // loop through sorted scores within currentRange from updated score pointer and update its
-      // corresponding appearance frequency in aggregatedValues
+      // corresponding appearance frequency
       int scoreIdx;
-      for (scoreIdx = updatingScoreIdx; scoreIdx < sentimentScores.size(); scoreIdx++) {
-        BigDecimal scorePoint = BigDecimal.valueOf(sentimentScores.get(scoreIdx));
-        if ((scorePoint.compareTo(nextPoint) < 0) || nextPoint.compareTo(UPPER_END) == 0) {
-          aggregateValues.put(currentRange, aggregateValues.get(currentRange) + 1);
+      for (scoreIdx = updatingScoreIdx; scoreIdx < userCommentList.size(); scoreIdx++) {
+        BigDecimal scorePoint = BigDecimal.valueOf(userCommentList.get(scoreIdx).getScore());
+        if ((scorePoint.compareTo(nextPoint) < 0) || nextPoint.compareTo(upperEnd) == 0) {
+          currentFrequency += 1;
+          // TODO: add topNumComments to the priority queue highMagnitudeList
         } else {
           break;
         }
       }
       // update the score pointer
       updatingScoreIdx = scoreIdx;
+      sentimentBucketList.add(
+          new SentimentBucket(
+              new ArrayList<>(highMagnitudeComments), currentFrequency, currentRange));
```

Comment:
While this doesn't matter now since we're not actually putting anything in highMagnitudeComments, we should be careful with converting the `PriorityQueue` to a list in this way.

If we want to preserve the priorityQueue ordering, we'll need to take advantage of the `poll` method of the priority queue.